File: Python/Bit Manipulation/Q5_6.py
def swapbits(a):
    bitarr = list(bin(a)[2:])
    for i in xrange(0, len(bitarr), 2):
        if i < len(bitarr) - 1:
            temp = bitarr[i]
            bitarr[i] = bitarr[i + 1]
            bitarr[i + 1] = temp
    print "".join(bitarr)


def main():
    swapbits(123)

if __name__ in "__main__":
    main()

File: Java/Hired Programming Practice/src/BitManipulation/Q5_6.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package BitManipulation;

import java.util.Arrays;

/**
 *
 * @author marcsantiago
 */
public class Q5_6 {
    public static void main(String[] args){
        swapBits(29);
        System.out.println(Integer.toBinaryString(123));
        swapBits(123);

    }
    
    public static void swapBits(int a){
        char [] bitArr = Integer.toBinaryString(a).toCharArray();
        for (int i = 0; i < bitArr.length; i = i + 2) {
            if(i < bitArr.length -1){
                char temp = bitArr[i];
                bitArr[i] = bitArr[i + 1];
                bitArr[i + 1] = temp;
            }
        }
        System.out.println(Arrays.toString(bitArr));
    }
    

}
